packages/oas-resolver/index.js

```diff
@@ -139,29 +140,53 @@ function testProtocol(input, backup) {
     return 'file:';
 }
 
-function resolveExternal(root, pointer, options, callback) {
-    var u = url.parse(options.source);
-    var base = options.source.split('\\').join('/').split('/');
+function getAbsoluteRefPath(source, $ref) {
+    let base = source.split('\\').join('/').split('/');
     let doc = base.pop(); // drop the actual filename
     if (!doc) base.pop(); // in case it ended with a /
+    base = base.join('/');
+    let u = url.parse(source);
+    let u2 = url.parse($ref);
+    let effectiveProtocol = testProtocol(u2.protocol, u.protocol)
+    if (effectiveProtocol === 'file:') {
+        return {
+            target: path.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+    else {
+        return {
+            target: url.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+}
+
+function resolveToName(ref, refs) {
+    let currentListOfResolved = [];
+    for (let key in refs) {
+        if (refs[key].resolvedAt) currentListOfResolved.push(refs[key].resolvedAt);
+    }
+
+    let potentialSchemaName = `#/components/schemas/${path.parse(path.basename(ref)).name}`;
```

Comment:
What happened to providing an optional overriding function passed into the options?

Reply:
We discussed and determined that if the default implementation handles conflicts then we would not really need an override function so why offer it because it would make things more complex that is not needed right now.  Thoughts?

Reply:
I agree that it serves our needs. But we also want the core maintainers to take the feature. The ability to override lets external users customize the behavior without resorting to forking and modifying the code base. I thought that would make this feature more versatile and therefore more acceptable as a general feature.

Reply:
We decided to wait on adding the overriding function for now as this meet our needs.